stan/math/opencl/kernel_generator/multi_result_kernel.hpp

```diff
@@ -0,0 +1,425 @@
+        std::set<const operation_cl_base*>& generated, name_generator& ng,
+        const std::string& i, const std::string& j,
+        const std::tuple<wrapper<T_results>...>& results,
+        const std::tuple<wrapper<T_expressions>...>& expressions) {
+      kernel_parts parts
+          = next::generate(generated, ng, i, j, results, expressions);
+      if (is_no_output<T_current_expression>::value) {
+        return parts;
+      }
+      kernel_parts parts0
+          = std::get<n>(expressions)
+                .x.get_whole_kernel_parts(generated, ng, i, j,
+                                          std::get<n>(results).x);
+      parts.initialization += parts0.initialization;
+      parts.body += parts0.body;
+      parts.reduction += parts0.reduction;
+      parts.args += parts0.args;
+      return parts;
+    }
+
+    static void set_args(
+        std::set<const operation_cl_base*>& generated, cl::Kernel& kernel,
+        int& arg_num, const std::tuple<wrapper<T_results>...>& results,
+        const std::tuple<wrapper<T_expressions>...>& expressions) {
+      next::set_args(generated, kernel, arg_num, results, expressions);
+
+      if (is_no_output<T_current_expression>::value) {
+        return;
+      }
+
+      std::get<n>(expressions).x.set_args(generated, kernel, arg_num);
+      std::get<n>(results).x.set_args(generated, kernel, arg_num);
+    }
+
+    static void add_event(
+        cl::Event e, const std::tuple<wrapper<T_results>...>& results,
+        const std::tuple<wrapper<T_expressions>...>& expressions) {
+      next::add_event(e, results, expressions);
+
+      std::get<n>(expressions).x.add_read_event(e);
+      std::get<n>(results).x.add_write_event(e);
+    }
+  };
+};
+
+template <typename... T_results>
+struct multi_result_kernel_internal<-1, T_results...> {
+  template <typename... T_expressions>
+  struct inner {
+    static void check_assign_dimensions(
+        int n_rows, int n_cols,
+        const std::tuple<wrapper<T_results>...>& results,
+        const std::tuple<wrapper<T_expressions>...>& expressions) {
+      return;
+    }
+
+    static kernel_parts generate(
+        std::set<const operation_cl_base*>& generated, name_generator& ng,
+        const std::string& i, const std::string& j,
+        const std::tuple<wrapper<T_results>...>& results,
+        const std::tuple<wrapper<T_expressions>...>& expressions) {
+      return {};
+    }
+
+    static void set_args(
+        std::set<const operation_cl_base*>& generated, cl::Kernel& kernel,
+        int& arg_num, const std::tuple<wrapper<T_results>...>& results,
+        const std::tuple<wrapper<T_expressions>...>& expressions) {
+      return;
+    }
+
+    static void add_event(
+        cl::Event e, const std::tuple<wrapper<T_results>...>& results,
+        const std::tuple<wrapper<T_expressions>...>& expressions) {
+      return;
+    }
+  };
+};
+
+template <int n, typename... T_results>
+template <typename... T_expressions>
```

Comment:
Missing comment

Reply:
This is just a definition of a member variable. Why does it need a comment?

Reply:
Ah yeah, nevermind.